google/cloud/pubsub_v1/publisher/client.py

```diff
@@ -372,6 +392,7 @@ def publish(self, topic, data, ordering_key="", **attrs):
 
             # Delegate the publishing to the sequencer.
             future = sequencer.publish(message)
+            future.add_done_callback(on_publish_done)
```

Comment:
Looking at the [futures code](https://github.com/googleapis/python-pubsub/blob/master/google/cloud/pubsub_v1/futures.py#L147-L186), done callbacks are called both on future success and error (they both call `_trigger()`), thus the message always gets released from the flow controller.

This also holds for already done futures, since [add_done_callback()](https://github.com/googleapis/python-pubsub/blob/master/google/cloud/pubsub_v1/futures.py#L131-L145) just invokes the callback immediately in this case - or am I missing something?